README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# Rokemon <img src="man/figures/logo.png" align="right" height="125px"/>

An R package to create Pokemon inspired ggplots. It also comes with
dataset of 801 Pokemon with 41 different features (Gotta analyze’em
all\!).

## Overview

For more details and examples see next sections.

Data

  - `pokemon`: Data frame containing attributes and stats of 801
    Pokemon.

Functions

  - `gghealth()`: HP bar inspired Bar charts.
  - `poke_pie()`: create pie charts from color distribution of Pokemon
    sprites.

Themes

  - `theme_rocket()`: Team Rocket theme
  - `theme_gameboy()` and `theme_gba()`: classic Gameboy and Gameboy
    Advanced themes
  - `theme_status()`: inspired by Pokemon status bar
  - `theme_mystic()`, `theme_valor()`, `theme_instinct()`: Pokemon Go
    teams theme; work well with `annotate_pogo()`
  - `scale_color_poketype()` and `scale_fill_poketype()`: Provides
    colors, if Pokemon types are mapped to color/fill

Pokemon Palettes

  - `poke_pal()`: color palettes created from Pokemon sprites
  - `display_poke_pal()`: view a Pokemon color palette

## Install

The developer version can be obtained from github.

``` r
#install.packages("devtools")
devtools::install_github("schochastics/Rokemon")
```

``` r
library(Rokemon)
library(tidyverse)

data(pokemon)
```

## Themes

The package includes several themes for ggplot.

### Theme Rocket

*(See what I did there…)*

``` r
ggplot(pokemon,aes(attack,defense))+
  geom_point(col = "grey")+
  theme_rocket()+
  labs(x = "Jessy",y = "James",
       title = "Theme Rocket",
       subtitle = "blast off at the speed of light!",
       caption = "meowth that's right")
```

<img src="figures/theme-rocket-1.png" width="80%" style="display: block; margin: auto;" />

### Gamyboy theme

If you want to get nostalgic.

``` r
ggplot(pokemon,aes(attack,defense))+
  geom_point(shape = 15,col = "#006400",size=2)+
  theme_gameboy()+
  labs(title = "Classic Gameboy Theme")
```

<img src="figures/theme-gameboy-1.png" width="80%" style="display: block; margin: auto;" />

``` r
ggplot(pokemon,aes(attack,defense))+
  geom_point(shape = 15,col = "#27408B",size=2)+
  theme_gba()+
  labs(title = "Gameboy Advanced Theme")
```

<img src="figures/theme-gba-1.png" width="80%" style="display: block; margin: auto;" />

### Status theme and HP bar chart

A theme inspired by HP bar in older Pokemon games. The theme is used in
`gghealth`, a function that plots bar charts in HP bar style.

``` r
pokemon[1:10,] %>% 
  gghealth("name","base_total",init.size = 5)+
  labs(x="",y="Stats Total")
```

<img src="figures/gghealth-1.png" width="80%" style="display: block; margin: auto;" />

### Pokemon Go

Annotate your plots with the logo of your favorite Pokémon Go team.

``` r

p1 <- pokemon %>%
  dplyr::filter(type1=="water") %>%
  ggplot(aes(defense,attack))+geom_point()+annotate_pogo(team = "mystic")+theme_mystic()+
  labs(title="Team Mystic",subtitle="Water Pokemon")

p2 <- pokemon %>%
  dplyr::filter(type1=="fire") %>%
  ggplot(aes(defense,attack))+geom_point()+annotate_pogo(team = "valor")+theme_valor()+
  labs(title="Team Valor",subtitle="Fire Pokemon")

p3 <- pokemon %>%
  dplyr::filter(type1=="electric") %>%
  ggplot(aes(defense,attack))+geom_point()+annotate_pogo(team = "instinct")+theme_instinct()+
  labs(title="Team Instinct",subtitle="Electric Pokemon")

gridExtra::grid.arrange(grobs=list(p1,p2,p3),ncol=3)
```

<img src="figures/pogo-teams-1.png" width="80%" style="display: block; margin: auto;" />

## Poke Pie

Create pie charts of the color distribution of Pokemon sprites. Download
all sprites, for example from
[here](https://github.com/PokeAPI/sprites).

``` r
#basic usage
poke_pie(path_to_sprites,pokemon_name)
```

![](figures/poke-pies.png)

The function is a reimplementation of
[this](https://gist.github.com/need12648430/4d681c9d1b18745ce159) code,
which was posted on
[reddit](https://www.reddit.com/r/pokemon/comments/2ey1pw/last_night_i_wrote_a_processing_script_that/ck45c21/)
a while ago.

## Color Palettes

The package also includes color palettes, which were automatically
generated from all 801 pokemon sprites.

``` r
poke_pal(name,n)
display_poke_pal(name)
```

![](figures/palettes.png)

Additionally there is also a palette Pokemon Types, used by
`scale_color_poketype()` and `scale_fill_poketype()`.
![](figures/rocket-type-pal.png)

I did not check all Pokemon palettes, so there may well be some
meaningless ones. A better alternative would be to use the dedicated
package `palettetown`. See the github
[repo](https://github.comt/imcdlucas/palettetown) for help.

``` r
install.packages('palettetown')
```

## Fonts

The package uses an old school gameboy font for some of its themes,
which can be download
[here](https://github.com/Superpencil/pokemon-font/releases/tag/v1.8.1).

In order to use the font in R you need the `extrafont` package.

``` r
#install.packages("extrafont")
extrafont::font_import() #only run ones
extrafont::loadfonts()
```

Alternatively, you can use the function `import_pokefont()`.

``` r
import_pokefont()
```

## Example use of data

Using `theme_rocket()` to create an effectiveness table of Pokemon
types.

``` r
pokemon %>%
  distinct(type1,.keep_all=TRUE) %>%
  select(defender = type1,against_bug:against_water) %>%
  gather(attacker,effect,against_bug:against_water) %>%
  mutate(attacker = str_replace_all(attacker,"against_",""))  %>%
  ggplot(aes(y=attacker,x=defender,fill=factor(effect)))+
  geom_tile()+
  geom_text(aes(label=ifelse(effect!=1,effect,"")))+
  scale_fill_manual(values=c("#8B1A1A", "#CD2626", "#EE2C2C", "#FFFFFF", "#00CD00", "#008B00"))+
  theme_rocket(legend.position="none")+
  labs(title="Effectiveness Table")
```

<img src="figures/effectiveness-1.png" width="80%" style="display: block; margin: auto;" />

Using Pokemon type colors

``` r
ggplot(pokemon,aes(defense,attack))+
  geom_point(aes(col=type1))+
  scale_color_poketype()+
  theme_bw()
```

<img src="figures/poketype-colors-1.png" width="80%" style="display: block; margin: auto;" />

## Addendum

  - Logo created by [ZeroDawn0D](https://github.com/ZeroDawn0D)
  - Pogo Logos downloaded
    [here](https://dribbble.com/shots/2831980-Pok-mon-GO-Team-Logos-Vector-Download)
  - Pokémon data download from
    [Kaggle](https://www.kaggle.com/rounakbanik/pokemon), originally
    scraped from [serebii.net](http://serebii.net/)
  - Sprites for `poke_pie` can be found
    [here](https://github.com/PokeAPI/sprites)


NEWS.md
# Rokemon 0.1.0





R/zzz.R
.onAttach <- function(libname, pkgname) {

  # if (interactive()) {
  #   packageStartupMessage(paste0("This package is in an early stage.\n",
  #                                "Check out https://github.com/schochastics/Rokemon for developments"))
  # }


  if (.Platform$OS.type == "windows")  { # nocov start
    if (interactive()) packageStartupMessage("Registering Windows fonts with R")
    extrafont::loadfonts("win", quiet = TRUE)
  }

  # if (getOption("poke.loadfonts", default = FALSE)) {
  #   if (interactive()) packageStartupMessage("Registering PDF & PostScript fonts with R")
  #   extrafont::loadfonts("pdf", quiet = TRUE)
  #   extrafont::loadfonts("postscript", quiet = TRUE)
  # }

  fnt <- extrafont::fonttable()
  if (!all(c("pokemon-font")%in% fnt$FamilyName)) {
    packageStartupMessage("NOTE: Pokemon font is required to use this package.")
    packageStartupMessage("      Please use Rokemon::import_pokefont() to install it")
  } # nocov end

}



R/poke_pie.R
#' Poke Pie Charts
#'
#'Create Pie Charts of the color distribution of Pokemon sprites.
#'
#'@param path_to_sprites path to sprite files. See Details
#'@param poke name or pokedex number of Pokemon
#'
#'@details Sprites are not provided and have to be download prior of usage. It is important
#'that the sprites are numbered from 1.png to 801.png. \url{https://github.com/PokeAPI/sprites}
#' contains all sprites in the right format and has been used for testing.
#'
#'@export
poke_pie <- function(path_to_sprites="",poke){
  if(is.character(poke)){
    poke <- tolower(poke)
    poke <- gsub(" ","",poke)
    poke <- gsub("\\.","",poke)
    poke <- gsub("-","",poke)

    if(poke=="nidoran"){
      warning("no gender specified for Nidoran. Using female as default\n")
      poke <- "nidoranf"
    }

    if (!(poke %in% pokenames)) {
      warning(paste(poke, "does not seem to be a Pokemon or is spelled wrongly\n. Using Pikachu instead.\n"))
      poke <- "pikachu"
    }
    poke <- which(pokenames==poke)
  }

  sprite <- png::readPNG(paste0(path_to_sprites,poke,".png"))
  dimension <- dim(sprite)
  img_rgb <- data.frame(
    x = rep(1:dimension[2], each = dimension[1]),
    y = rep(dimension[1]:1, dimension[2]),
    R = as.vector(sprite[,,1]),
    G = as.vector(sprite[,,2]),
    B = as.vector(sprite[,,3])
  )
  img_tbl <- dplyr::count(dplyr::group_by_(img_rgb,"R","G","B"))
  img_tbl <- img_tbl[img_tbl$R!=0 | img_tbl$G!=0 | img_tbl$B!=0,]
  img_tbl <- dplyr::ungroup(img_tbl[img_tbl$R!=1 | img_tbl$G!=1 | img_tbl$B!=1,])
  img_tbl <- dplyr::arrange_(img_tbl,"n")
  img_tbl$col <- grDevices::rgb(img_tbl$R,img_tbl$G,img_tbl$B)

  n <- nrow(img_tbl)
  img_tbl$t <- 1:n

  img <- grid::rasterGrob(sprite, interpolate=TRUE)

  p <- ggplot2::ggplot(img_tbl,ggplot2::aes(x="",y=n,fill=factor(t)))+
    ggplot2::geom_col()+
    ggplot2::coord_polar("y",start=0)+
    ggplot2::scale_fill_manual(values=img_tbl$col)+ggplot2::theme_minimal()+
    ggplot2::theme(legend.position = "none",
          panel.grid=ggplot2::element_blank(),
          axis.text=ggplot2::element_blank(),
          axis.title=ggplot2::element_blank())
  q <- cowplot::ggdraw(p)+ cowplot::draw_grob(img,0.3,0.3,0.4,0.4)

  q

}


R/healthbar.R
# GeomHealthbar <- ggproto("GeomHealthbar",Geom,
#                          required_aes = c('x','y'),
#                          default_aes = aes(size = 10),
#                          draw_key = draw_key_abline,
#                          draw_panel = function(data, panel_scales,coord){
#                            coords <- coord$transform(data, panel_scales)
#                            grid::segmentsGrob(x0 = coords$x,x1 = coords$x,
#                                               y0 = 0,y1 = 1,
#                                               default.units = "native",
#                                               gp = grid::gpar(colour="yellow",size = coords$size,lineend="round"))#+
#                            # grid::segmentsGrob(x0 = coord$x,x1 = coord$x,
#                            #                    y0 = 0,y1 = 1,default.units = "npc",
#                            #                    gp = grid::gpar(colour="white",size = 2,lineend="round"))+
#                            # grid::segmentsGrob(x0 = coord$x,x1 = coord$x,
#                            #                    y0 = 0,y1 = coord$y,default.units = "npc",
#                            #                    gp = grid::gpar(colour="green",size = 1,lineend="square"))
#                          }
# )
#
# geom_healthbar <-  function(mapping = NULL, data = NULL, stat = "identity",
#                             position = "identity", show.legend = NA,
#                             inherit.aes = TRUE, ...) {
#   layer(
#     geom = GeomHealthbar, mapping = mapping, data = data, stat = stat,
#     position = position, show.legend = show.legend, inherit.aes = inherit.aes,
#     params = list(...)
#   )
# }
#
# pokemon %>% .[1:50,] %>%
#   mutate(defense=defense/max(defense)) %>%
#   ggplot(aes(x=reorder(name,defense)))+geom_healthbar(aes(y=defense))+coord_flip()

#' HP bar chart
#'
#'A barchart in the style of a HP bar.
#'
#'@param data data.frame.
#'@param names column name of names to use in the plot.
#'@param values column name of values to plot.
#'@param sort.names order names according to value (TRUE, default) or not (FALSE).
#'@param init.size size of the healthbar. See details.
#'@param cuts numeric vector of length 3. See details.
#'
#'@details \code{init.size} controls the height of the bar. The more names are used, the smaller
#'the value should be. The values in \code{cuts} should be decreasing in the interval (0,1).
#'
#'@export
#'@examples
#'\dontrun{
#'gghealth(pokemon[1:10,],"name","base_total",init.size = 5)+
#'  labs(x="",y="Stats Total")
#'}
gghealth <- function(data,names,values,sort.names=TRUE,
                     init.size=3,cuts = c(0.8,0.4,0.2)){

  fracs <- (data[[values]]-min(data[[values]]))/(max(data[[values]])-min(data[[values]]))
  max_val <- max(data[[values]])
  init.size <- max(init.size,3)

  if(sort.names){
    ordered <- order(data[[values]])
  } else{
    ordered <- 1:nrow(data)
  }

  data[[names]] <- factor(data[[names]],levels = data[[names]][ordered])
  hp_col <- 0
  hp_col[fracs<cuts[3]] <- 4
  hp_col[fracs>=cuts[3]] <- 3
  hp_col[fracs>=cuts[2]] <- 2
  hp_col[fracs>=cuts[1]] <- 1

  data[["hp_col"]] <- factor(hp_col,levels=1:4)

  p <- ggplot2::ggplot(data,ggplot2::aes_string(x = names))+
    ggplot2::geom_segment(ggplot2::aes_string(xend = names),yend = max_val,y = 0,
                     size = init.size,lineend = "round")+
    ggplot2::geom_segment(ggplot2::aes_string(xend = names),yend = max_val,y = 0,
                     size = init.size-1,lineend = "round",colour = "white")+
    ggplot2::geom_segment(ggplot2::aes_string(xend = names,yend = values, colour = "hp_col"),
                     y = 0,size = init.size-1,lineend = "butt")+
    ggplot2::scale_colour_manual(values = c(`1`="#66CD00",`2`="#609869",`3`="#FB9400",`4`="#FF4400"))+
    ggplot2::coord_flip()+
        theme_status()
  p
}




R/theme_gameboy.R
#' Classic Gamyboy inspired ggplot theme
#'
#' Theme based on the first generation of the Gameboy.
#'
#' @export
#' @examples
#' \dontrun{
#' library(ggplot2)
#'
#' ggplot(pokemon,aes(attack,defense))+
#'  geom_point(shape = 15,col = c("#006400"),size=2)+
#'  theme_gameboy()+
#'  labs(title = "Classic Gameboy Theme")
#'}
theme_gameboy <- function() {
  title_font <- "pokemon-font"

  gb <- ggplot2::theme(panel.grid = ggplot2::element_blank(),
          # panel.grid.major = ggplot2::element_line(colour="#008B00"),
          panel.background = ggplot2::element_rect(fill="#79BC00"),
          panel.border = ggplot2::element_rect(fill=NA,colour="#8F8B8F",size=4),
          plot.background = ggplot2::element_rect(fill="#C1BDB7"),
          text = ggplot2::element_text(family = "pokemon-font"),
          plot.title = ggplot2::element_text(family = title_font,colour="#561E94"),
          plot.subtitle = ggplot2::element_text(family = title_font,colour="#561E94"),
          axis.ticks = ggplot2::element_line(colour="#8F8B8F"),
          legend.background = ggplot2::element_rect(fill="#C1BDB7"),
          legend.position = "bottom",
          legend.text = ggplot2::element_text(colour="#561E94"),
          legend.key = ggplot2::element_rect(fill="#C1BDB7",colour="black"),
          plot.caption = ggplot2::element_text(colour = "#A20055"))
  return(gb)
}
#-------------------------------------------------------------------------------
#' Gamyboy Advanced inspired ggplot theme
#'
#' Theme based on the Gameboy Advanced.
#'
#' @export
#' @examples
#' \dontrun{
#' library(ggplot2)
#'
#' ggplot(pokemon,aes(attack,defense))+
#'  geom_point(shape = 15,col = c("#006400"),size=2)+
#'  theme_gameboy()+
#'  labs(title = "Classic Gameboy Theme")
#'}
theme_gba <- function() {
  title_font <- "pokemon-font"

  gb <- ggplot2::theme(panel.grid = ggplot2::element_blank(),
               panel.background = ggplot2::element_rect(fill="#00BFFF"),
               panel.border = ggplot2::element_rect(fill=NA,colour="black",size=4),
               plot.background = ggplot2::element_rect(fill="#68228B"),
               text = ggplot2::element_text(family = "pokemon-font"),
               plot.title = ggplot2::element_text(family = title_font,colour="black"),
               plot.subtitle = ggplot2::element_text(family = title_font,colour="black"),
               axis.ticks = ggplot2::element_line(colour="black"),
               axis.text = ggplot2::element_text(colour="#8F8B8F"),
               axis.title = ggplot2::element_text(colour="#8F8B8F"),
               legend.background = ggplot2::element_rect(fill="#68228B"),
               legend.position = "bottom",
               legend.text = ggplot2::element_text(colour="#8F8B8F"),
               legend.key = ggplot2::element_rect(fill="#68228B",colour="black"),
               plot.caption = ggplot2::element_text(colour = "#8F8B8F"))
  return(gb)
}

#-------------------------------------------------------------------------------
#' Health bar theme
#'
#' Theme used in gghealth.
#'
#'
#' @export
theme_status <- function(){
    hp <- ggplot2::theme(legend.position="none",
            axis.line = ggplot2::element_line(size = 1,lineend = "square"),
            axis.text.y = ggplot2::element_text(family = "pokemon-font",colour = "#ecdb6d"),
            axis.text.x = ggplot2::element_text(family = "pokemon-font",colour = "#ecdb6d"),
            axis.title = ggplot2::element_text(family = "pokemon-font",colour = "#ecdb6d"),
            plot.title = ggplot2::element_text(family = "pokemon-font",colour = "#ecdb6d"),
            plot.subtitle = ggplot2::element_text(family = "pokemon-font",colour = "#ecdb6d"),
            plot.caption = ggplot2::element_text(family = "pokemon-font",colour = "#ecdb6d"),
            axis.ticks.x = ggplot2::element_line(colour = "#ecdb6d"),
            axis.ticks.y = ggplot2::element_line(colour = "#ecdb6d"),
            plot.background = ggplot2::element_rect(fill = "black"),
            panel.background = ggplot2::element_rect(fill = "white"),
            panel.grid = ggplot2::element_blank(),
            strip.background = ggplot2::element_rect(fill="#ecdb6d"),
            strip.text = ggplot2::element_text(family = "pokemon-font",colour = "black"))
    return(hp)
}

#-------------------------------------------------------------------------------
#' Theme Rocket
#'
#' ggplot theme of Team Rocket. Meowth that's right!
#'
#' @param ... additional parameters to ggplot2::theme()
#'
#' @export
#' @examples
#' \dontrun{
#' #create a Pokemon type effectiveness chart
#' library(tidyverse)
#'
#' pokemon %>%
#'   distinct(type1,.keep_all=TRUE) %>%
#'   select(defender = type1,against_bug:against_water) %>%
#'   gather(attacker,effect,against_bug:against_water) %>%
#'   mutate(attacker = str_replace_all(attacker,"against_",""))  %>%
#'   ggplot(aes(y=attacker,x=defender,fill=factor(effect)))+
#'   geom_tile()+
#'   geom_text(aes(label=ifelse(effect!=1,effect,"")))+
#'   scale_fill_manual(values=c("#8B1A1A", "#CD2626", "#EE2C2C", "#FFFFFF", "#00CD00", "#008B00"))+
#'   theme_rocket(legend.position="none")+
#'   labs(title="Effectiveness Table")
#' }
theme_rocket <- function(...){
  rocket <- ggplot2::theme(panel.background = ggplot2::element_rect(fill="#FFFFFF",colour=NA),
        plot.background =  ggplot2::element_rect(fill="#0F0F0F",colour=NA),
        panel.grid.minor = ggplot2::element_blank(),
        panel.grid.major = ggplot2::element_line(colour="#DEBD39",linetype = 8),
        axis.title.x = ggplot2::element_text(colour = "#EE30A7"),
        axis.text.x = ggplot2::element_text(colour = "#EE30A7"),
        axis.line.x = ggplot2::element_line(colour = "#EE30A7",size=1.5,lineend = "square"),
        axis.ticks.x = ggplot2::element_line(colour = "#EE30A7"),
        axis.title.y = ggplot2::element_text(colour = "#8968CD"),
        axis.text.y = ggplot2::element_text(colour = "#8968CD"),
        axis.line.y = ggplot2::element_line(colour = "#8968CD",size=1.5,lineend = "square"),
        axis.ticks.y = ggplot2::element_line(colour = "#8968CD"),
        plot.title = ggplot2::element_text(colour="#DEBD39"),
        plot.subtitle = ggplot2::element_text(colour="#DEBD39"),
        plot.caption = ggplot2::element_text(colour="#DEBD39"),
        strip.background = ggplot2::element_rect(fill="#0F0F0F",colour=NA),
        strip.text = ggplot2::element_text(colour="#DEBD39"),
        legend.background = ggplot2::element_rect(fill="#0F0F0F"),
        legend.text = ggplot2::element_text(colour="#DEBD39"),
        legend.title = ggplot2::element_text(colour="#DEBD39",face = "bold"),
        legend.key = ggplot2::element_rect(fill="#0F0F0F",colour="#DEBD39"),...)

  return(rocket)
}
#-------------------------------------------------------------------------------
#' Pokemon Go Team Logos
#'
#' Enhance your plots with a background picture of your favorite Pokemon Go team.
#'
#'@param team string. either 'mystic', 'valor' or 'instinct'
#'@param opacity opacity of team logo.
#'
#'@references The logos were not designed by me. All credit goes to
#'\href{https://dribbble.com/shots/2831980-Pok-mon-GO-Team-Logos-Vector-Download}{this guy}.
#'
#' @export
#' @examples
#' \dontrun{
#' library(tidyverse)
#'
#' pokemon %>%
#'  dplyr::filter(type1=="water") %>%
#'  ggplot(aes(defense,attack))+
#'  annotate_pogo(team = "mystic")+
#'  geom_point()+
#'  theme_mystic()+
#'  labs(title="Team Mystic",subtitle="Water Pokemon")
#' }
annotate_pogo <- function(team="mystic",opacity=0.5){
  if(!team%in%c("mystic","valor","instinct")){
    stop("team must be one of 'mystic', 'valor','instinct'")
  }
  bg_dir <- system.file("backgrounds",package="Rokemon")
  img <- png::readPNG(paste0(bg_dir,"/team-",team,".png"))
  img[,,4] <- opacity*img[,,4]
  g <- grid::rasterGrob(img, interpolate=TRUE)

  pogo <- ggplot2::annotation_custom(g, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf)

  return(pogo)
}
#-------------------------------------------------------------------------------
#' Theme Mystic
#'
#' ggplot theme that simply adds a blue background to the plot.
#'
#' @export
theme_mystic <- function(){
  mystic <- ggplot2::theme(panel.background = ggplot2::element_rect(fill="white"),
                           plot.background = ggplot2::element_rect(fill="#4B38F7"),
                           panel.grid = ggplot2::element_blank(),
                           plot.title = ggplot2::element_text(colour="white"),
                           plot.subtitle = ggplot2::element_text(colour="white"),
                           plot.caption = ggplot2::element_text(colour="white"),
                           strip.background = ggplot2::element_rect(fill="#483D8B"),
                           strip.text = ggplot2::element_text(colour="white"),
                           axis.text = ggplot2::element_text(colour="white"),
                           axis.ticks = ggplot2::element_line(colour="white"),
                           axis.title = ggplot2::element_text(colour="white")
  )
  return(mystic)
}

#-------------------------------------------------------------------------------
#' Theme Valor
#'
#' ggplot theme that simply adds a red background to the plot.
#'
#' @export
theme_valor <- function(){
  valor <- ggplot2::theme(panel.background = ggplot2::element_rect(fill="white"),
                          plot.background = ggplot2::element_rect(fill="#FC0000"),
                          panel.grid = ggplot2::element_blank(),
                          strip.background = ggplot2::element_rect(fill="#CD0000")
  )
  return(valor)
}

#-------------------------------------------------------------------------------
#' Theme Instinct
#'
#' ggplot theme that simply adds a yellow background to the plot.
#'
#' @export
theme_instinct <- function(){
  instinct <- ggplot2::theme(panel.background = ggplot2::element_rect(fill="white"),
                             plot.background = ggplot2::element_rect(fill="#F7E300"),
                             panel.grid = ggplot2::element_blank(),
                             strip.background = ggplot2::element_rect(fill="#FFD700")
  )
  return(instinct)
}

#-------------------------------------------------------------------------------
#'Pokemon color scale
#'
#' This allows to use the Pokemon type color scale in ggplot.
#'
#'@param ... common discrete scale parameters for ggplot
#'@export
#'@examples
#'library(ggplot2)
#'
#'ggplot(pokemon,aes(defense,attack))+
#'   geom_point(aes(col=type1))+
#'   scale_color_poketype()
#'
#'ggplot(pokemon,aes(type1))+
#'   geom_bar(aes(fill=type1))+
#'   scale_color_poketype()

scale_color_poketype <- function(...){
  type_pal <- c(`normal`= "#A8A77A",`fire`= "#EE8130",`water`= "#6390F0",
              `electric`= "#F7D02C",`grass`= "#7AC74C",`ice`= "#96D9D6",
              `fighting`= "#C22E28",`poison`= "#A33EA1",`ground`= "#E2BF65",
                `flying`= "#A98FF3",`psychic`= "#F95587",`bug`= "#A6B91A",
                  `rock`= "#B6A136",`ghost`= "#735797",`dragon`= "#6F35FC",
                  `dark`= "#705746",`steel`= "#B7B7CE",`fairy`= "#D685AD")

  ggplot2::scale_color_manual(values = type_pal,...)
}

#' @rdname scale_color_poketype
#' @export
scale_fill_poketype <- function(...){
  type_pal <- c(`normal`= "#A8A77A",`fire`= "#EE8130",`water`= "#6390F0",
                `electric`= "#F7D02C",`grass`= "#7AC74C",`ice`= "#96D9D6",
                `fighting`= "#C22E28",`poison`= "#A33EA1",`ground`= "#E2BF65",
                `flying`= "#A98FF3",`psychic`= "#F95587",`bug`= "#A6B91A",
                `rock`= "#B6A136",`ghost`= "#735797",`dragon`= "#6F35FC",
                `dark`= "#705746",`steel`= "#B7B7CE",`fairy`= "#D685AD")

  ggplot2::scale_fill_manual(values = type_pal,...)
}
